src/controllers/serviceaccounts.go

```
@@ -96,3 +136,55 @@ func (c *serviceAccountController) getServiceAccountToken(serviceAccount *corev1
 
 	return "", customerrors.NewValidationError(fmt.Sprintf(ErrNoTokenFound, serviceAccount.Name))
 }
+
+// CreateServiceAccount creates a new ServiceAccount with the given name and namespace.
+func (c *serviceAccountController) CreateServiceAccount(name, namespace string) (types.ServiceAccount, error) {
+	serviceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				utils.ManagedLabel: utils.ManagedLabelValue,
+			},
+		},
+	}
+	_, err := c.client.CoreV1().ServiceAccounts(namespace).Create(c.ctx, serviceAccount, metav1.CreateOptions{})
+	if err != nil {
+		return types.ServiceAccount{}, customerrors.NewAPIError(fmt.Sprintf(ErrCouldNotCreateServiceAccount, name, namespace), err)
+	}
+	return types.ServiceAccount{Name: name}, nil
+}
+
+// DeleteServiceAccount deletes a ServiceAccount by name and namespace.
+func (c *serviceAccountController) DeleteServiceAccount(name, namespace string) (types.MessageResponse, error) {
+	err := c.client.CoreV1().ServiceAccounts(namespace).Delete(c.ctx, name, metav1.DeleteOptions{})
+	if err != nil {
+		return types.MessageResponse{}, customerrors.NewAPIError(fmt.Sprintf(ErrCouldNotDeleteServiceAccount, name, namespace), err)
+	}
+
+	return types.MessageResponse{}, nil
```

You can just return an error here without an empty type

---

Or maybe return the serviceAccount name here as well